### DISCUSSION TITLE
Upgrading from 0.7 to 0.8: `Ref::new_slice` gone?

### DISCUSSION THREAD

**Discussion by ChristopherRabotin** [google/zerocopy#1974] [Q&A] [answered]
Hi there,

After reading through the [upgrading guide in the announcement](https://github.com/google/zerocopy/discussions/1680) and searching in the docs, I can't seem to find a method equivalent to 0.7's [`Ref::new_slice`](https://docs.rs/zerocopy/0.7.35/zerocopy/struct.Ref.html#method.new_slice). Using the new `from_bytes` seems like it should work at first thought, but there is no way to convert the underlying bytes into a slice of type `T`, like [I've done here](https://github.com/nyx-space/anise/blob/9a1c2a285fd212e7228f536dce07b099ddcd3d21/anise/src/naif/daf/daf.rs#L160-L166):

```rust
// (...)
        Ok(match Ref::new_slice(&rcrd_bytes[SummaryRecord::SIZE..]) {
            Some(data) => data.into_slice(),
            None => &{
                R::default();
                [] as [R; 0]
            },
        })
```

Has this functionality been moved to another struct, or has it been removed entirely?

Thanks

**Comment by joshlf** [comment, marked as answer]
Given a `Ref<[T]>`, both `Ref::from_bytes` and `Ref::into_ref` will work as expected - you no longer need to special-case slices either during construction or when converting to a vanilla reference.

**Reply by ChristopherRabotin** [reply]
Thanks!

The following should be the equivalent in version 0.8:

```rust
        Ok(
            match Ref::<_, [R]>::from_bytes(&rcrd_bytes[SummaryRecord::SIZE..]) {
                Ok(r) => Ref::into_ref(r),
                Err(_) => &{
                    R::default();
                    [] as [R; 0]
                },
            },
        )
```
